### PR TITLE
add multi-model stake_set support for NumerAPI #70

### PR DESCRIPTION
**Pull request by npezolano**

### CODE REVIEW

File: numerapi/numerapi.py [hunk 1]
```diff
@@ -734,11 +734,12 @@ def get_leaderboard(self, limit: int = 50, offset: int = 0) -> List[Dict]:
             utils.replace(item, "nmrStaked", utils.parse_float_string)
         return data
 
-    def stake_set(self, nmr) -> Dict:
+    def stake_set(self, nmr, username: str = None) -> Dict:
```
Comment (uuazed) [inline numerapi/numerapi.py]: please change that to `model_id`, similar to the other endpoints and make it a required parameter

File: numerapi/numerapi.py [hunk 2]
```diff
@@ -765,7 +766,8 @@ def stake_set(self, nmr) -> Dict:
              'value': '10'}
         """
         # get username of logged in user
-        username = self.get_account()['username']
+        if not username:
+            username = self.get_account()['username']
```
Comment (uuazed) [inline numerapi/numerapi.py]: this part won*t be needed after switching to model_id

**Review by uuazed — CHANGES_REQUESTED** [review]
Thanks for the PR! I prefer to work with the model_id and make it a required parameter. Multi-model accounts are the standard nowadays.

File: numerapi/numerapi.py [hunk 3]
```diff
@@ -777,9 +776,9 @@ def stake_set(self, nmr) -> Dict:
             nmr = decimal.Decimal(str(nmr))
         # update stake!
         if nmr < current:
-            return self.stake_decrease(current - nmr)
+            return self.stake_decrease(current - nmr, username)
```
Comment (uuazed) [inline numerapi/numerapi.py]: username is undefined

Comment (npezolano) [reply numerapi/numerapi.py]: Sorry missed that one let me fix it

**Pull request merged**